application/dash_application/utility/plot_builder.py

```diff
@@ -791,4 +794,241 @@ def items_for_df(self):
             PINERY_COL.TissuePreparation: self.tissue_materials,
             COMMON_COL.Reference: self.reference,
-        }
\ No newline at end of file
+        }
+
+
+class Subplot:
+    def __init__(
+            self,
+            title,
+            df,
+            x_col,
+            y_col,
+            y_label,
+            colourby,
+            shapeby,
+            hovertext_cols,
+            cutoff_lines,
+            markermode,
+            bar_positive,
+            bar_negative,
+            log_y
+    ):
+        self.title = title
+        self.y_label = y_label
+        self.df = df
+        self.x_col = x_col
+        self.y_col = y_col
+        self.colourby = colourby
+        self.shapeby = shapeby
+        self.hovertext_cols = hovertext_cols
+        self.cutoff_lines = [] if cutoff_lines is None else cutoff_lines
+        self.markermode = markermode
+        self.bar_positive = bar_positive
+        self.bar_negative = bar_negative
+        self.log_y = log_y
+
+    def traces(self):
+        return _generate_traces(
+            self.df,
+            self.x_col,
+            self.y_col,
+            self.colourby,
+            self.shapeby,
+            self.hovertext_cols,
+            self.cutoff_lines,
+            self.markermode,
+            self.bar_positive,
+            self.bar_negative,
+        )
+
+
+class SingleLaneSubplot(Subplot):
+    def __init__(
+            self,
+            title,
+            df,
+            x_col,
+            y_col,
+            y_label,
+            colourby,
+            shapeby,
+            hovertext_cols,
+            cutoff_lines=None,
+            markermode="markers",
+            bar_positive=None,
+            bar_negative=None,
+            log_y=False
+    ):
+        super().__init__(
+            title,
+            df,
+            x_col,
+            y_col,
+            y_label,
+            colourby,
+            shapeby,
+            hovertext_cols,
+            cutoff_lines,
+            markermode,
+            bar_positive,
+            bar_negative,
+            log_y,
+        )
+
+
+class CallReadySubplot(Subplot):
+    def __init__(
+            self,
+            title,
+            df,
+            x_col,
+            y_col,
+            y_label,
+            colourby,
+            shapeby,
+            hovertext_cols,
+            cutoff_lines=None,
+            markermode="markers",
+            bar_positive=None,
+            bar_negative=None,
+            log_y=False,
+    ):
+        super().__init__(
+            title,
+            df,
+            x_col,
+            y_col,
+            y_label,
+            colourby,
+            shapeby,
+            hovertext_cols,
+            cutoff_lines,
+            markermode,
+            bar_positive,
+            bar_negative,
+            log_y,
+        )
+
+
+# TODO: Currently not in use. Big question: How to deal with titles in Dashi?
+class GraphTitles:
```

Comment:
I would prefer if you deleted this and we can bring it back when we're ready to use it.